ulid_test.go

```diff
@@ -399,6 +399,46 @@ func TestEntropy(t *testing.T) {
 	}
 }
 
+type flakyEntropyReader struct {
+	data []byte
+	pos  int
+}
+
+func (r *flakyEntropyReader) Read(p []byte) (int, error) {
+	remaining := len(r.data) - r.pos
+	if remaining <= 0 {
+		return 0, io.EOF
+	}
+
+	n := rand.Intn(len(p)) + 1
```

Comment:
Do we need the randomness to make this flaky? Perhaps simply always reading less than `len(p)` would suffice.

Reply:
Changed, now reading only a half of the required size.